libminifi/include/core/state/nodes/AgentInformation.h

```diff
@@ -623,18 +632,70 @@ class AgentManifest : public DeviceInformation {
 };
 
 /**
- * Purpose and Justification: Prints classes along with their properties for the current agent.
+ * This class is used for regular heartbeat without manifest
+ * A light weight heartbeat
  */
-class AgentInformation : public DeviceInformation, public AgentMonitor, public AgentIdentifier {
+class AgentInformationWithoutManifest : public DeviceInformation, public AgentMonitor, public AgentIdentifier {
+public:
+
+    AgentInformationWithoutManifest(std::string name, utils::Identifier & uuid)
+    : DeviceInformation(name, uuid) {
+      setArray(false);
+ }
+
+    AgentInformationWithoutManifest(const std::string &name)
```

Comment:
Single parameter constructors should either be marked `explicit` or have a comment explaining why implicit conversion from the parameter type is desired.

Same with `AgentInformationWithManifest`.

Reply:
fair point. Thanks for catching it.